### PR TITLE
Revert "chore(deps): bump golang.org/x/crypto"

### PR DESCRIPTION
**Pull request by ibuziuk**
https://issues.redhat.com/browse/CRW-4642
This reverts commit 311c85612da80b795d5ec18dcd0c3e99149fd91d.

Verified via `rm -fr vendor;  go mod tidy -go 1.15; go mod vendor; make build`

```
PASS
ok  	github.com/che-incubator/kubernetes-image-puller/utils	(cached)
CGO_ENABLED=1 GOOS=linux go build -v -o ./bin/kubernetes-image-puller ./cmd/main.go
CGO_ENABLED=1 GOOS=linux go build -a -ldflags '-w -s' -a -installsuffix cgo -o ./bin/sleep ./sleep/sleep.go
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,27 +4,26 @@ go 1.15
...
 	github.com/imdario/mergo v0.3.8 // indirect
-	github.com/json-iterator/go v1.1.6 // indirect
-	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
-	github.com/modern-go/reflect2 v1.0.1 // indirect
 	github.com/peterbourgon/diskv v2.0.1+incompatible // indirect
+	github.com/pkg/errors v0.9.1 // indirect
 	github.com/spf13/pflag v1.0.5 // indirect
 	github.com/stretchr/testify v1.4.0
-	golang.org/x/crypto v0.1.0 // indirect
+	golang.org/x/crypto v0.0.0-20180808211826-de0752318171 // indirect
```
Comment (nickboldt) [inline go.mod:17]: Problem with this is that you're removing the (old version 0.1.0](https://github.com/che-incubator/kubernetes-image-puller/pull/155/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6L20-R17) from Oct 2022 ([latest is 0.11 from Jul 5 2023](https://pkg.go.dev/golang.org/x/crypto?tab=versions)) and going back to a version from 2018. That seems like a bad choice, given this is a crypto package and we just spent 2 weeks fighting with crypto issues for FIPS compliance. I'll have a look today to see if the KIP and KIPO are FIPS-compliant ... if they're not, then we may have to fix them or else state that the imagepuller doesn't work on FIPS clusters.

Comment (ibuziuk) [reply go.mod:17]: @nickboldt whatever works, please go ahead and update the modules to be compliant with brew builds

Comment (nickboldt) [reply go.mod:17]: FYI https://github.com/eclipse/che/issues/22416

**Comment by nickboldt** [comment]
If I update the go.mod to 1.19:

```
rm -fr vendor/; go mod tidy -go 1.19; go mod vendor; make build
```

```
# github.com/che-incubator/kubernetes-image-puller/utils
utils/clusterutils.go:57:67: not enough arguments in call to clientset.AppsV1().DaemonSets(cfg.Namespace).Watch
	have ("k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions)
	want (context.Context, "k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions)
utils/clusterutils.go:59:3: unknown field 'IncludeUninitialized' in struct literal of type "k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions
utils/clusterutils.go:74:87: not enough arguments in call to clientset.AppsV1().Deployments(cfg.Namespace).Get
	have (string, "k8s.io/apimachinery/pkg/apis/meta/v1".GetOptions)
	want (context.Context, string, "k8s.io/apimachinery/pkg/apis/meta/v1".GetOptions)
utils/clusterutils.go:152:64: not enough arguments in call to clientset.AppsV1().DaemonSets(cfg.Namespace).Create
	have (*"k8s.io/api/apps/v1".DaemonSet)
	want (context.Context, *"k8s.io/api/apps/v1".DaemonSet, "k8s.io/apimachinery/pkg/apis/meta/v1".CreateOptions)
utils/clusterutils.go:196:82: not enough arguments in call to clientset.AppsV1().DaemonSets(cfg.Namespace).Get
	have (string, "k8s.io/apimachinery/pkg/apis/meta/v1".GetOptions)
	want (context.Context, string, "k8s.io/apimachinery/pkg/apis/meta/v1".GetOptions)
utils/clusterutils.go:227:80: not enough arguments in call to clientset.AppsV1().DaemonSets(cfg.Namespace).Delete
	have (string, *"k8s.io/apimachinery/pkg/apis/meta/v1".DeleteOptions)
	want (context.Context, string, "k8s.io/apimachinery/pkg/apis/meta/v1".DeleteOptions)
utils/operations.go:55:27: not enough arguments in call to clientset.AppsV1().DaemonSets(cfg.Namespace).Get
	have (string, "k8s.io/apimachinery/pkg/apis/meta/v1".GetOptions)
	want (context.Context, string, "k8s.io/apimachinery/pkg/apis/meta/v1".GetOptions)
utils/operations.go:72:27: not enough arguments in call to clientset.AppsV1().DaemonSets(cfg.Namespace).Get
	have (string, "k8s.io/apimachinery/pkg/apis/meta/v1".GetOptions)
	want (context.Context, string, "k8s.io/apimachinery/pkg/apis/meta/v1".GetOptions)
utils/operations.go:93:27: not enough arguments in call to clientset.AppsV1().DaemonSets(cfg.Namespace).Get
	have (string, "k8s.io/apimachinery/pkg/apis/meta/v1".GetOptions)
	want (context.Context, string, "k8s.io/apimachinery/pkg/apis/meta/v1".GetOptions)
```

Does that help figure out what changes need to be done in utils/ ?

File: go.mod (hunk 1 above)
Comment (ibuziuk) [reply go.mod:17]: > That seems like a bad choice, given this is a crypto package and we just spent 2 weeks fighting with crypto issues for FIPS compliance.

this particular commit with version bump was added to the main Jul 27 and was not part of the r FIPS compliance exercise. Feel free to close it or update if needed but it is the very same version of image puller that was packaged for 3.7 / 3.7.1

**Review by SDawley — APPROVED** [review]
Per https://issues.redhat.com/browse/CRW-4781 , we'll revert the commit and eventually stop building downstream.

**Pull request merged**